src/valkey-cli.c

```diff
@@ -1610,17 +1610,23 @@ static int cliSwitchProto(void) {
     return result;
 }
 
+static void resetConfig(void) {
+    config.dbnum = 0;
+    config.in_multi = 0;
+    config.pubsub_mode = 0;
+}
+
 /* Connect to the server. It is possible to pass certain flags to the function:
  *      CC_FORCE: The connection is performed even if there is already
  *                a connected socket.
  *      CC_QUIET: Don't print errors if connection fails. */
 static int cliConnect(int flags) {
+    resetConfig();
+
     if (context == NULL || flags & CC_FORCE) {
         if (context != NULL) {
```

Comment:
If there's already a connection and FORCE is not given, we don't want to reset the config, do we?

```suggestion
static int cliConnect(int flags) {
    if (context == NULL || flags & CC_FORCE) {
        resetConfig();
        if (context != NULL) {
```